module/execution/scripts.go

```diff
@@ -107,26 +70,35 @@ func NewScripts(
 	)
 
 	return &Scripts{
-		executor:         queryExecutor,
-		headers:          header,
-		registerAtHeight: registerAtHeight,
+		log:               zerolog.New(log).With().Str("component", "script_executor").Logger(),
+		executor:          queryExecutor,
+		headers:           header,
+		compatibleHeights: NewCompatibleHeights(log, versionControl, minHeight, maxHeight),
 	}
 }
 
 // ExecuteAtBlockHeight executes provided script against the block height.
 // A result value is returned encoded as byte array. An error will be returned if script
 // doesn't successfully execute.
-// Expected errors:
-// - Script execution related errors
```

Comment:
now that we're less relaxed about error handling, i think we need to add one of the fvm catchall error types to make sure request related script errors are handled.

specifically, we should document `fvmerrors.CodedError`, which is any user fvm error that is not fatal.

Reply:
looking at the error handling, I think we already handle these properly for script execution. we just need to adjust the error docs